cranelift/codegen/src/isa/x64/inst/args.rs

```diff
@@ -186,89 +181,185 @@ pub enum RMI_R_Op {
+            | SseOpcode::Cvtsi2sd
+            | SseOpcode::Cvtss2sd
+            | SseOpcode::Cvttsd2si
+            | SseOpcode::Divsd
+            | SseOpcode::Maxsd
+            | SseOpcode::Minsd
+            | SseOpcode::Movsd
+            | SseOpcode::Mulsd
+            | SseOpcode::Sqrtsd
+            | SseOpcode::Subsd
+            | SseOpcode::Ucomisd
+            | SseOpcode::Comisd => SSE2,
+
+            SseOpcode::Roundss | SseOpcode::Roundsd => SSE41,
+        }
+    }
+
     pub(crate) fn to_string(&self) -> String {
         match self {
-            SSE_Op::SSE_Addss => "addss".to_string(),
-            SSE_Op::SSE_Subss => "subss".to_string(),
-            SSE_Op::SSE_Movss => "movss".to_string(),
-            SSE_Op::SSE2_Movsd => "movsd".to_string(),
+            SseOpcode::Addss => "addss".to_string(),
+            SseOpcode::Subss => "subss".to_string(),
+            SseOpcode::Movss => "movss".to_string(),
+            SseOpcode::Movsd => "movsd".to_string(),
             _ => "unimplemented sse_op".to_string(),
         }
     }
 }
 
-impl fmt::Debug for SSE_Op {
+impl fmt::Debug for SseOpcode {
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        write!(fmt, "{}", self.to_string())
+        let name = match self {
+            SseOpcode::Addss => "addss",
+            SseOpcode::Addsd => "addsd",
+            SseOpcode::Comiss => "comiss",
+            SseOpcode::Comisd => "comisd",
+            SseOpcode::Cvtsd2ss => "cvtsd2ss",
+            SseOpcode::Cvtsd2si => "cvtsd2si",
+            SseOpcode::Cvtsi2ss => "cvtsi2ss",
+            SseOpcode::Cvtsi2sd => "cvtsi2sd",
+            SseOpcode::Cvtss2si => "cvtss2si",
+            SseOpcode::Cvtss2sd => "cvtss2sd",
+            SseOpcode::Cvttss2si => "cvttss2si",
+            SseOpcode::Cvttsd2si => "cvttsd2si",
+            SseOpcode::Divss => "divss",
+            SseOpcode::Divsd => "divsd",
+            SseOpcode::Maxss => "maxss",
+            SseOpcode::Maxsd => "maxsd",
+            SseOpcode::Minss => "minss",
+            SseOpcode::Minsd => "minsd",
+            SseOpcode::Movss => "movss",
+            SseOpcode::Movsd => "movsd",
+            SseOpcode::Mulss => "mulss",
+            SseOpcode::Mulsd => "mulsd",
+            SseOpcode::Rcpss => "rcpss",
+            SseOpcode::Roundss => "roundss",
+            SseOpcode::Roundsd => "roundsd",
+            SseOpcode::Rsqrtss => "rsqrtss",
+            SseOpcode::Sqrtss => "srtqss",
+            SseOpcode::Sqrtsd => "sqrtsd",
+            SseOpcode::Subss => "subss",
+            SseOpcode::Subsd => "subsd",
+            SseOpcode::Ucomiss => "ucomiss",
+            SseOpcode::Ucomisd => "ucomisd",
+        };
+        write!(fmt, "{}", name)
+    }
+}
+
+impl ToString for SseOpcode {
+    fn to_string(&self) -> String {
+        format!("{:?}", self)
+    }
+}
+
+/// Some SSE operations requiring 3 operands i, r/m, and r.
+#[derive(Clone, PartialEq)]
+pub enum SseRmiOpcode {
```

Comment:
Hmm .. I'm not entirely happy about the `i` in `Rmi` here.  Those immediates are often an opcode extension, especially for the comparison instructions, and not really an operand.  By contrast, the `i` in `AluRmi` really is an operand.  Hence there's a conceptual inconsistency.  Not sure what a better name is though.

Reply:
I see. What about `SseRmIOpcode` (with capital I), to indicate it is its own operand? Or in a more verbose way, `SseRmImmOpcode`?

Reply:
That still doesn't make clear that the imm isn't an operand.  How about `SseRmExtendedOpcode` ?

Reply:
Actually .. thinking about this more .. I don't think there is any point in having these be separate from the plain `SseOpcode`.  They are "just more" SSE* instructions and the only difference is that the encoding of the instruction has an extra byte, which IIRC is always right at the end of the instruction.  So I think you should merge these into `SseOpcode` and delete `SseRmiOpcode`.  (w/ apologies for changing my story on this)